clients/src/main/java/org/apache/kafka/clients/admin/AbortTransactionSpec.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.admin;
+
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.annotation.InterfaceStability;
+
+import java.util.Objects;
+
+@InterfaceStability.Evolving
+public class AbortTransactionSpec {
+    private final TopicPartition topicPartition;
+    private final long producerId;
+    private final short producerEpoch;
+    private final int coordinatorEpoch;
```

Comment:
In the KIP, you also mentioned `transactionStartOffset`. Is it not required anymore or do you plan to add it later on?

Reply:
Yeah, I am planning to leave it for later. I recall running into some complication when I tried to implement the broker-side changes for this, but I don't remember specifically what it was. In any case, I think it can come after the tool is checked in since we need the old API for compatibility anyway.

Reply:
Sounds good. Do we already have a Jira to track this?

Reply:
Created this one: https://issues.apache.org/jira/browse/KAFKA-12725.